targets/esp8266/source/jerry_extapi.c

```diff
@@ -16,7 +16,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 
-#include "jerry-core/jerryscript.h"
+#include "jerry-core/include/jerryscript.h"
```

Comment:
This is "user error". The build system of ESP8266 should add "jerry-core/include" to the include path and then `#include "jerryscript.h"` only. Actually, even before this PR, the ESP8266 build system should have added "jerry-core" to the include path and never _ever_ have anything like `#include "jerry-core/xxxxx"` in the code. But I did not want to go into rewriting the build systems of any of targets. (Where the build system was correctly written, no changes in the code was necessary -- see particle, riot-stm32f4, zephyr.)